crates/wysiwyg/src/dom/nodes/container_node.rs

```diff
@@ -405,39 +405,42 @@ where
     /// Returns false if no updates was done.
     /// e.g. first text-like node is already a ZWSP.
     pub fn add_leading_zwsp(&mut self) -> bool {
+        fn insert_zwsp<S>(container: &mut ContainerNode<S>, handle_is_set: bool)
+        where
+            S: UnicodeString,
+        {
+            if handle_is_set {
+                container.insert_child(0, DomNode::new_zwsp());
+            } else {
+                container.children.insert(0, DomNode::new_zwsp());
+            }
+        }
+
         // Note: handle might not be set in cases where we are transforming a node
         // that is detached from the DOM. In that case it's fine to transform it
         // without worrying about handles.
         let handle_is_set = self.handle().is_set();
         let Some(first_child) = self.children.get_mut(0) else {
-            return false;
+            insert_zwsp(self, handle_is_set);
+            return true;
         };
         match first_child {
             DomNode::Container(c) => c.add_leading_zwsp(),
             DomNode::Zwsp(_) => false,
             DomNode::Text(t) => {
-                match (handle_is_set, t.data().is_empty()) {
-                    (true, true) => {
+                if t.data().is_empty() {
+                    if handle_is_set {
                         self.replace_child(0, vec![DomNode::new_zwsp()]);
-                    }
-                    (true, false) => {
-                        self.insert_child(0, DomNode::new_zwsp());
-                    }
-                    (false, true) => {
+                    } else {
                         self.children[0] = DomNode::new_zwsp();
                     }
-                    (false, false) => {
-                        self.children.insert(0, DomNode::new_zwsp());
-                    }
+                } else {
+                    insert_zwsp(self, handle_is_set);
                 }
                 true
             }
             DomNode::LineBreak(_) => {
-                if handle_is_set {
-                    self.insert_child(0, DomNode::new_zwsp());
-                } else {
-                    self.children.insert(0, DomNode::new_zwsp());
-                }
+                insert_zwsp(self, handle_is_set);
```

Comment:
Again not a necessary change, but I _think_ we could use a match guard here to help clarify what branches result in the same control flow.

Disclaimer, this may not work. Could we do something like:

```
DomNode::Text(t) if t.data.is_empty => ...,
DomNode::Text(_) | DomNode::Linebreak(_) => {
    insert_zwsp(...);
    true
}

```

Think it would help cut down on a level of indentation and make it clearer that in non empty text cases it's treated the same as a linebreak.

Reply:
Yes, definitely. I guess I'm not used yet to work with conditional cases on Rust matches 😅